examples/gpu-sharing/model/model.py

```diff
@@ -0,0 +1,85 @@
+import threading
+import time
+from typing import Any, Dict, List
+
+import torch
+
+from .img_img_model import Model as ImgImgModel
+from .inpaint_model import Model as InpaintModel
+from .text_img_model import Model as TextImgModel
+
+MODEL_SUB_NAME_ARG = "model_sub_name"
+
+
+class Model:
+    def __init__(self, **kwargs) -> None:
+        self._models = {
+            "img_img": ImgImgModel(**kwargs),
+            "text_img": TextImgModel(**kwargs),
+            "inpaint": InpaintModel(**kwargs),
+        }
+        self._current_prediction_model_sub_name = None
+        self._prediction_lock = threading.Lock()
+
+    def load(self):
+        for model_name, model in self._models.items():
+            print(f"Loading model {model_name}")
+            model.load()
+            # Keep model in cpu memory, but evict from GPU memory
+            self.unload()
+            print(f"Model {model_name} loaded successfully")
+
+    def unload(self):
+        torch.cuda.empty_cache()
```

Comment:
@aspctu @bolasim I don't think this unloads the model from GPU memory. Is there a better way? Is there a way to check that this is unloading model from GPU memory? e.g. is there a way to get GPU memory usage programmatically? If so then I could log memory usage before and after to double check.
The example works because I think all three models fit in GPU memory on A10G at the same time.

Reply:
To unload the model, we have to move it to CPU before emptying the cache.
So here are the two functions I used in my experiments that worked well. `prepare` loads the model in GPU. `standby` takes it out of the GPU.
```
    def prepare(self):
        self._model.to("cuda")

    def standby(self):
        self._model.to("cpu")
```

It's best to force CUDA to recompute the memory before clearing that cache or loading anything else into the GPU. You can do that by using this function 
```
        # Force re-calc GPU memory
        torch.cuda.memory_allocated()
```
Let me know if this helps.